rewrite-java/src/main/java/org/openrewrite/java/typegeneration/AsmUtils.java

```diff
@@ -0,0 +1,361 @@
+
+        //Interfaces *ClassTypeSignature
+        if (classType.getInterfaces() != null && !classType.getInterfaces().isEmpty()) {
+            for (JavaType interfaceType : classType.getInterfaces()) {
+                String interfaceSig = getClassTypeSignature(TypeUtils.asFullyQualified(interfaceType));
+                if (interfaceSig == null) return null;
+                signature.append(interfaceSig);
+            }
+        }
+        return signature.toString();
+    }
+
+    /**
+     * This is used for generating an ASM method signature from a JavaType.Method.
+     *
+     * <method.typeParameters>(method.parameters)method.returnType^method.thrownExceptions
+     *
+     * @param method Rewrite's method type
+     * @return The asm signature for the method type or null if there is any missing type information.
+     */
+    @Nullable
+    public static String getMethodTypeSignature(@Nullable JavaType.Method method) {
+
+        //TypeParams? "(" TypeSignature* ")" (TypeSignature | V) Exception
+
+        if (method == null || (method.getGenericSignature() == null && method.getResolvedSignature() == null)) {
+            return null;
+        }
+        JavaType.Method.Signature methodSignature = method.getResolvedSignature() != null ? method.getResolvedSignature() : method.getGenericSignature();
+
+        StringBuilder signature = new StringBuilder();
+
+        //TypeParams?
+        if (method.getTypeParameters() != null && !method.getTypeParameters().isEmpty()) {
+            signature.append("<");
+            for (JavaType typeParameter :  method.getTypeParameters()) {
+                if (typeParameter instanceof JavaType.GenericTypeVariable) {
+                    JavaType.GenericTypeVariable generic = (JavaType.GenericTypeVariable) typeParameter;
+                    signature.append(generic.getFullyQualifiedName())
+                            .append("::")
+                            .append((generic.getBound() == null) ? "Ljava/lang/Object;" : getFieldTypeSignature(generic.getBound()));
+                } else {
+                    return null;
+                }
+            }
+            signature.append(">");
+        }
+        signature.append('(');
+        //Method Parameters : TypeSignature*
+        for (JavaType methodParameterType : methodSignature.getParamTypes()) {
+            String methodParameterSig = getTypeSignature(methodParameterType);
+            if (methodParameterSig == null) {
+                return null;
+            }
+            signature.append(methodParameterSig);
+        }
+        signature.append(')');
+
+        //Return Type can be either : TypeSignature | V
+        if (method.getName().equals("<init>") || methodSignature.getReturnType() == null) {
+            signature.append("V");
+        } else {
+            String typeSig = getTypeSignature(methodSignature.getReturnType());
+            if (typeSig == null) return null;
```

Comment:
I don't like omitting the curly braces because it becomes easy to write code like this:
```java
if(condition)
    i++;
    j++;
```
Where a human reader could easily misinterpret the `j++` as being part of the conditional

```suggestion
            if (typeSig == null) {
                return null;
            }
```

Reply:
fixed all places I was doing this.